### PR TITLE
Added Utils module, tests, travis config.

### PR DESCRIPTION
**Pull request by n1k0**
This is WiP. This introduces a new `Utils` module (yes, I know) containing a `timeAgo` helper which I'm planning to use to render human friendly times in views. This also introduces unit tests for this specific function, and a test runner based on `elm-test`, as well as Travis-CI integration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
 # Kinto Elm Experiments
 
 Experimenting with [Elm](elm-lang.org) and [Kinto](http://www.kinto-storage.org/).
+
+## Tests
+
+```
+$ npm i
```
Comment (magopian) [inline README.md:8]: i'd rather have the full version here, what do you think?

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "build": "node_modules/.bin/elm-make src/Main.elm --output=build/app.js",
     "publish-to-gh-pages": "npm run build && cp html/index.html build/ && node_modules/.bin/gh-pages --dist build/",
-    "test": "echo \"Testing is cheating!\" && exit 1"
+    "test": "node_modules/.bin/elm-test"
```
Comment (magopian) [inline package.json:9]: oh so testing is not cheating after all, eh? <3

File: tests/elm-package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        "../src"
+    ],
+    "exposed-modules": [],
+    "dependencies": {
+        "elm-lang/core": "4.0.0 <= v < 5.0.0",
+        "elm-community/elm-test": "2.0.0 <= v < 3.0.0",
+        "rtfeldman/node-test-runner": "2.0.0 <= v < 3.0.0"
+    },
+    "elm-version": "0.17.0 <= v < 0.18.0"
+}
```
Comment (magopian) [inline tests/elm-package.json:17]: I believe this file has to be manually synchronized with the one from the root folder of the project, which might raise issues in the future. I wonder if there's a way to automatize that one way or another? (check point 4 in the bullet points list down the page on http://package.elm-lang.org/packages/elm-community/elm-test/latest):

> Copy all the dependencies from elm-package.json into tests/elm-package.json. These dependencies need to stay in sync, so make sure whenever you change your dependencies in your current elm-package.json, you make the same change to tests/elm-package.json.

**Comment by n1k0** [comment]
Note that a `elm-test --watch` option is [coming](https://github.com/rtfeldman/node-test-runner/commit/9def5ff7942218d52296005e288abe91f955745c) in its next release :)

File: src/View.elm [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
 import Model exposing (Model, Record, Msg(..))
 import Form
 
 
-formatLatsModified : Float -> String
-formatLatsModified timestamp =
-    format "%Y-%m-%d@%H:%M" (Date.fromTime timestamp)
+formatLastModified : Int -> Time -> String
+formatLastModified timestamp currentTime =
+    timeAgo (toFloat timestamp) currentTime
```
Comment (magopian) [inline src/View.elm:14]: I don't understand the `toFloat` here, is it to convert the `Int` to something that is "compatible" with a `Time`? From https://github.com/n1k0/kinto-elm-experiments/pull/2/files#diff-05d2065a68c705b72271bf365cdc74eeR6 i thought `timeAgo` was taking a Time, not a float?!

File: src/Utils.elm [hunk 5]
```diff
@@ -0,0 +1,64 @@
+module Utils exposing (timeAgo)
+
+import Time exposing (Time)
+
+
+timeAgo : Time -> Time -> String
+timeAgo time now =
+    let
+        seconds =
+            round ((now - time) / 1000)
```
Comment (magopian) [inline src/Utils.elm]: Do you also want an integer division here, instead of using `round`?

File: src/Utils.elm [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            minutes // 60
+
+        days =
+            hours // 24
+
+        weeks =
+            days // 7
+
+        months =
+            days // 30
```
Comment (magopian) [inline src/Utils.elm]: You use `30` for a month here, but you test with `31`. Should the test be on `30` too?

File: tests/Tests.elm [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        , timeAgoTest "<1h diff" (60 * 30) "30 minutes ago"
+        , timeAgoTest "=1h diff" (60 * 60) "an hour ago"
+        , timeAgoTest "<1d diff" (60 * 60 * 12) "12 hours ago"
+        , timeAgoTest "=1d diff" (60 * 60 * 24) "a day ago"
+        , timeAgoTest "<1month diff" (60 * 60 * 24 * 12) "12 days ago"
+        , timeAgoTest "=1month diff" (60 * 60 * 24 * 31) "a month ago"
+        , timeAgoTest "<1month diff, week match" (60 * 60 * 24 * 14) "2 weeks ago"
+        , timeAgoTest "<1year diff" (60 * 60 * 24 * 30 * 6) "6 months ago"
+        , timeAgoTest ">1 year" (60 * 60 * 24 * 365 * 12) "12 years ago"
+        ]
```
Comment (magopian) [inline tests/Tests.elm:53]: How about testing a few other corner cases? I'm thinking in particular of the following ones after reading the implementation:
- 2 hours ago (to test the plural of "hour")
- 28 days ago (to test that it's not "four weeks ago", but not "one month ago" either)
- maybe test corner cases where it's 59 seconds, 59 minutes, 23 hours, 6 days, 29 (or 30) days...

**Review by magopian — COMMENTED** [review]
This is a winner! Thanks Nico, I'm pretty sure this is going to be yet another blockbuster on github <3

I'm wondering about the "precision" we want. For example, if there's 6 days, should it say 6 days or "a week"? And if there's 29 days?
Also, if it's 8 days, should it be 8 days or "over a week ago" ?

**Comment by n1k0** [comment]
Current status: https://twitter.com/jackdre02/status/775957722470637568

File: src/View.elm (hunk 4 above)
Comment (n1k0) [reply src/View.elm:14]: I didn't get why this was required as `source-directories` includes `../src` and I thought the dependencies would be checked recursively, but I may be wrong. The thing is, tests pass, so they're definitely loaded appropriately. Now I don't quite know what to do with all this.

File: src/Utils.elm (hunk 5 above)
Comment (n1k0) [reply src/Utils.elm]: Let's work with floats everywhere and round numbers just before rendering them as strings.

File: src/Utils.elm (hunk 6 above)
Comment (n1k0) [reply src/Utils.elm]: Good catch.

**Comment by n1k0** [comment]
Addressed a bunch of things my way in https://github.com/n1k0/kinto-elm-experiments/pull/2/commits/d6c1bef814877dc858720db04d5cb91637672e6d and https://github.com/n1k0/kinto-elm-experiments/pull/2/commits/ab92c71db506f1ec7b4703b462db714bcb71cb82, feedback would be  appreciated :)

**Comment by magopian** [comment]
seems all good to me, congratz!
